### PR TITLE
Admin users did not work with custom user models

### PR DESCRIPTION
**Pull request by softr8**
# Description
If Spree.user_class was other than Spree::User, it was failing
to create edit and delete url's in Admin::UsersController#index,
using CustomUser for example, it tried to generate: admin_custom_user_url
instead of using the existing admin_user_url

# Checklist:
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [ ] I have updated Guides and README accordingly to this change (if needed)
- [x] I have added tests to cover this change (if needed)
- [ ] I have attached screenshots to this PR for visual changes (if needed)

### CODE REVIEW

File: backend/spec/controllers/spree/admin/users_controller_spec.rb [hunk 1]
```diff
@@ -59,6 +59,30 @@
...
     end
+
+    context "when Spree.user_class have a different namespace than Spree" do
+      class UserModel < ApplicationRecord
+        self.table_name = 'spree_users'
+        include Spree::UserMethods
+      end
+
+      before do
+        Spree.user_class = 'UserModel'
```
Comment (peterberkenbosch) [inline backend/spec/controllers/spree/admin/users_controller_spec.rb:70]: I think we should do an after block here too to make sure subsequent tests are using the original User.

**Review by jarednorman — APPROVED** [review]

File: backend/spec/controllers/spree/admin/users_controller_spec.rb [hunk 2]
```diff
@@ -59,6 +59,33 @@
...
+        self.table_name = 'spree_users'
+        include Spree::UserMethods
+      end
+
+      around do |example|
+        Spree.user_class = 'UserModel'
+        UserModel.create(email: "a@solidus.io")
+        example.run
+        Spree.user_class = 'Spree::LegacyUser'
+      end
```
Comment (waiting-for-dev) [inline backend/spec/controllers/spree/admin/users_controller_spec.rb:74]: Can we take the old value of `user_class` and reset it later instead of hardcoding the value that we are using now? I.e.:


```suggestion
        actual_user_class = Spree.user_class
        Spree.user_class = 'UserModel'
        UserModel.create(email: "a@solidus.io")
        example.run
        Spree.user_class = actual_user_class
      end
```

File: backend/app/views/spree/admin/store_credits/index.html.erb [hunk 3]
```diff
@@ -70,6 +70,6 @@
   <div class="no-objects-found">
     <%= render 'spree/admin/shared/no_objects_found',
                  resource: Spree::StoreCredit,
-                 new_resource_url: new_object_url %>
+                 new_resource_url: spree.new_admin_user_store_credit_url %>
```
Comment (waiting-for-dev) [inline backend/app/views/spree/admin/store_credits/index.html.erb:73]: I don't think it's bad to reference the URL helper directly, but it seems that the `new_object_url` shortcut is used extensively in the backend. Besides, it doesn't seem related to this PR. Can you help me understand why we need this change?

**Review by waiting-for-dev — CHANGES_REQUESTED** [review]

File: backend/app/views/spree/admin/store_credits/index.html.erb (hunk 3 above)
Comment (softr8) [reply backend/app/views/spree/admin/store_credits/index.html.erb:73]: it's the same issue than the one in users/index, it's a nested resource and it tries to build the url with two AR objects, so if the model that represents Spree.user_class is different than `user`, it does not match anymore with the routes.

`Spree.user_class = 'MyStore::Admin'` is going to generate new_admin_admin_store_credit_url, because it reads the model name, so, that's why we have to 'hardcode' the route name

Comment (waiting-for-dev) [reply backend/app/views/spree/admin/store_credits/index.html.erb:73]: Thanks!

File: backend/spec/controllers/spree/admin/users_controller_spec.rb [hunk 4]
```diff
@@ -59,6 +59,33 @@
...
     end
+
+    context "when Spree.user_class have a different namespace than Spree" do
+      class UserModel < ApplicationRecord
+        self.table_name = 'spree_users'
+        include Spree::UserMethods
+      end
+
+      around do |example|
+        actual_user_class = Spree.user_class
```
Comment (waiting-for-dev) [inline backend/spec/controllers/spree/admin/users_controller_spec.rb:70]: Oh, my fault. This should fix the CI, right?
```suggestion
        actual_user_class = Spree.user_class.name
```

**Review by waiting-for-dev — APPROVED** [review]
Thanks!!
